DCKUBE-629: Modified bitbucket elasticsearch example (#277)
atlassian/data-center-helm-charts: docs/docs/examples/elasticsearch/BITBUCKET_ELASTICSEARCH.md

```
then install it:
```bash
helm install elasticsearch --set image.tag="7.9.3" elastic/elasticsearch
```

### Configuring your Bitbucket deployment

> To enable using the installed Elasticsearch service you need to configure the service URL under `bitbucket:` in the `values.yaml` file:

Author: Updated.

> (Check Kubernetes official document on how to get [DNS for Services](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services){.external}).)

Author: Updated.

```
then install it:
```bash
helm install elasticsearch --set image.tag="7.9.3" elastic/elasticsearch
```

> > Running the above commands will install Elasticsearch with default configurations, i.e. 3 worker nodes. 
> > However, it may not always work out of the box if failed to fulfill pre-requisites for the default installation. 
> > Some example pre-requisites include:
> > * cpu/memory requests: 1000m/2Gi
> > * pre-configured storage with at least three 30Gi volumes (one for each worker node)
> > 
> > Refer to [Elasticsearch values.yaml file](https://github.com/elastic/helm-charts/blob/7.9/elasticsearch/values.yaml){.external} for more detail.

Reviewer: Since moving to mkdocs these callouts `>` should be replaced instead with an [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/)

for example:

```suggestion
!!!warning "Default configuration"
      
      Running the above commands will install Elasticsearch with default configurations, i.e. 3 worker nodes. However, it may not always work out of the box if failed to fulfill pre-requisites for the default installation.

      Pre-requisites include; cpu/memory requests: `1000m/2Gi`, storage with at least three `30Gi` volumes (one for each worker node)
      
      Refer to [Elasticsearch values.yaml file](https://github.com/elastic/helm-charts/blob/7.9/elasticsearch/values.yaml){.external} for more detail.
```

would be rendered as:

![image](https://user-images.githubusercontent.com/409063/131282914-3a8ef8ab-d658-4923-9425-a48919865be4.png)

### Configuring your Bitbucket deployment

To enable the installed Elasticsearch service you need to configure the service URL under `bitbucket:` in the `values.yaml` file.

> Check the Kubernetes official documentation on how to get [DNS for Services](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services){.external}).

Reviewer: There is still a trailing `)` brace which needs to be removed

```yaml
bitbucket:
  elasticSearch:
    baseUrl: http://elasticsearch-master.<namespace>.svc.cluster.local:9200
```
This will also have the effect of disabling Bitbucket’s internal Elasticsearch instance.

> > If you hold a basic license of Elasticsearch, authentication is not supported. However, if you have other [Elastic Stack subscriptions](https://www.elastic.co/subscriptions){.external}, you will also need to provide the details in a Kubernetes secret and configure that in the `values.yaml` file:

Reviewer: Same here, I think this should be replaced with an `!!!info`

```
then install it:
```bash

> helm install elasticsearch --set image.tag="7.9.3" elastic/elasticsearch

Reviewer 2: The standard `version` helm install parameter doesn't work for elasticsearch chart?

Author: No... It was causing error during the blitz. I think `version` is the helm chart version, while `image.tag` is the docker image version for Elasticsearch.

```
!!! info 

>     Running the above commands will install Elasticsearch with default configurations, i.e. 3 worker nodes. 

Reviewer 2: ```suggestion
    Running the above commands will install Elasticsearch with default configuration, i.e. 3 worker nodes. 
```

    However, it may not always work out of the box if failed to fulfill pre-requisites for the default installation. 
    Some example pre-requisites include:

>     * cpu/memory requests: 1000m/2Gi

Reviewer 2: ```suggestion
    * CPU/memory requests: 1000m/2Gi
```

>     * CPU/memory requests: 1000m/2Gi

Reviewer 3: I think CPU and Memory is for each worker. As the storage this is mentioned, is good to mention `for each worker` here to make it consistent.

>     * pre-configured storage with at least three 30Gi volumes (one for each worker node)

Reviewer 3: Maybe is cleaner to say `pre-configured storage with 30Gi volumes for each worker node.`

>     Refer to [Elasticsearch values.yaml file](https://github.com/elastic/helm-charts/blob/7.9/elasticsearch/values.yaml){.external} for more detail.

Reviewer 2: ```suggestion
    Refer to [Elasticsearch values.yaml file](https://github.com/elastic/helm-charts/blob/7.9/elasticsearch/values.yaml){.external} for more details.
```

### Configuring your Bitbucket deployment

> To enable the installed Elasticsearch service you need to configure the service URL under `bitbucket:` in the `values.yaml` file.

Reviewer 2: ```suggestion
To enable the installed Elasticsearch service you need to configure the service URL under `bitbucket:` stanza in the `values.yaml` file.
```

> Check the Kubernetes official documentation on how to get [DNS for Services](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services){.external}.

Reviewer 2: ```suggestion
Check the Kubernetes official documentation on how to get [DNS record for a service](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services){.external}.
```

```yaml
bitbucket:
  elasticSearch:
    baseUrl: http://elasticsearch-master.<namespace>.svc.cluster.local:9200
```
This will also have the effect of disabling Bitbucket’s internal Elasticsearch instance.

!!! info

>     If you hold a basic license of Elasticsearch, authentication is not supported. However, if you have other [Elastic Stack subscriptions](https://www.elastic.co/subscriptions){.external}, you will also need to provide the details in a Kubernetes secret and configure that in the `values.yaml` file:

Reviewer 2: There is no authentication with the basic license? Can you please share some resources about that?

      passwordSecretKey: password
```

> > Read about [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/){.external}.

Reviewer: Can we replace all of these callouts (`>`) with an appropriate [admonitions](https://squidfunk.github.io/mkdocs-material/reference/admonitions/) in this case this would be an `!!!info```

Reviewer: Think you forgot to do this one.

      passwordSecretKey: password
```
> Read about [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/){.external}.

## Testing your Elasticsearch connection

> To test if Elasticsearch is properly set up, go to **Administration > System - Server settings**. The Elasticsearch URL should be pre-populated already in the search section. Click the **Test** button to see if it connected successfully.

Reviewer 2: ```suggestion
To test if Elasticsearch is properly set up, go to **Administration > System - Server settings**. The Elasticsearch URL should be pre-populated already in the search section. Click the **Test** button to see if it connects successfully.
```
